### ISSUE TITLE
NullPointerException when Debug Launching Spring app

### ISSUE THREAD

**Issue by FieteO** [microsoft/vscode-spring-boot-dashboard#293]
When debug launching my spring app via the Spring Boot Dashboard, I got an error message and see a NullPointerException in the corresponding extension output. I assume it is related to the Memory usage view (which is working fine as far as I can tell).

# Output of the `vscode-spring-boot-debug-log`
```
Activating 'vscode-spring-boot' extension
"spring-boot.ls.java.home" setting not specified or empty value
Found java executable: /home/ubuntu/.vscode-server/extensions/redhat.java-1.16.0-linux-x64/jre/17.0.6-linux-x86_64/bin/java
isJavaEightOrHigher => true
Redirecting server logs to /dev/null
Disabling server log output. No more output will be sent after this.
[Error - 2:18:41 PM] Request workspace/executeCommand failed.
  Message: Internal error.
  Code: -32603 
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.springframework.ide.vscode.commons.languageserver.util.ParentProcessWatcher.lambda$apply$0(ParentProcessWatcher.java:129)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.reflect.InvocationTargetException
	at jdk.internal.reflect.GeneratedMethodAccessor22.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 12 more
Caused by: java.lang.NullPointerException: Cannot invoke "org.springframework.ide.vscode.boot.java.livehover.v2.SpringProcessMemoryMetricsLiveData.getHeapMemoryMetrics()" because "data" is null
	at org.springframework.ide.vscode.boot.java.livehover.v2.SpringProcessCommandHandler.handleLiveMetricsProcessRequest(SpringProcessCommandHandler.java:310)
	at org.springframework.ide.vscode.boot.java.livehover.v2.SpringProcessCommandHandler.lambda$new$5(SpringProcessCommandHandler.java:83)
	at org.springframework.ide.vscode.commons.languageserver.util.SimpleLanguageServer.executeCommand(SimpleLanguageServer.java:273)
	at org.springframework.ide.vscode.commons.languageserver.util.SimpleWorkspaceService.executeCommand(SimpleWorkspaceService.java:145)
	... 16 more
```
Note that I could not reliably reproduce this when restarting the debug session though.

**Comment by Eskibear** [comment]
Some clues from my observation while I was improving UX of memory view:
When live process is connected, the sequence of notifications (`onDidLiveProcessConnect` and the metrics related ones) received from vscode-spring-boot is not always the same. I assume it's because once onDidLiveProcessConnect is received, dashboard [starts querying memory data](https://github.com/microsoft/vscode-spring-boot-dashboard/blob/63f9ace351b75ded9ce4404e5319374012f54b3c/src/controllers/LiveDataController.ts#L72-L76), but the memory metrics live data object is not initialized in Spring LS yet, causing the error.

https://github.com/microsoft/vscode-spring-boot-dashboard/issues/287 and https://github.com/microsoft/vscode-spring-boot-dashboard/pull/274 might be related. And I believe it's not hard to fine tune the lifecycle and api calling sequence.

//cc @martinlippert @vudayani-vmw for awareness.

**Comment by Eskibear** [comment]
@FieteO It should be fixed by #294 . It will be shipped with the latest pre-release version in a few hours. You can give it a try by switching to pre-release version then.

**Comment by FieteO** [comment]
Thanks @Eskibear. I will check it out
